Makefile

```diff
@@ -82,176 +70,104 @@ $(MYGOBIN)/pluginator:
+	test-examples-kustomize-against-v4-release
 
-.PHONY: clean-kustomize-external-go-plugin
-clean-kustomize-external-go-plugin:
-	./hack/buildExternalGoPlugins.sh ./plugin clean
+# The following target referenced by a file in
+# https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/kustomize
+.PHONY: prow-presubmit-check
+prow-presubmit-check: \
+	install-tools \
+	test-unit-kustomize-plugins \
+	test-go-mod \
+	build-non-plugin-all \
+	test-examples-kustomize-against-HEAD \
+	test-examples-kustomize-against-v4-release
+
+.PHONY: license
+license: $(MYGOBIN)/addlicense
+	./hack/add-license.sh run
 
-### End kustomize plugin rules.
+.PHONY: check-license
+check-license: $(MYGOBIN)/addlicense
+	./hack/add-license.sh check
+
+.PHONY: lint
+lint: $(MYGOBIN)/golangci-lint $(MYGOBIN)/goimports $(builtinplugins)
+	./hack/for-each-module.sh "make lint"
 
 .PHONY: lint-kustomize
-lint-kustomize: $(MYGOBIN)/golangci-lint-kustomize $(builtinplugins)
-	cd api; $(MYGOBIN)/golangci-lint-kustomize \
-	  -c ../.golangci.yml \
-	  --path-prefix api \
-	  run ./...
-	cd kustomize; $(MYGOBIN)/golangci-lint-kustomize \
-	  -c ../.golangci.yml \
-	  --path-prefix kustomize \
-	  run ./...
-	cd cmd/pluginator; $(MYGOBIN)/golangci-lint-kustomize \
-	  -c ../../.golangci.yml \
-	  --path-prefix cmd/pluginator \
-	  run ./...
+lint-kustomize: $(MYGOBIN)/golangci-lint $(builtinplugins)
+	cd api; $(MAKE) lint
+	cd kustomize; $(MAKE) lint
+	cd cmd/pluginator; $(MAKE) lint
 
-# Used to add non-default compilation flags when experimenting with
-# plugin-to-api compatibility checks.
-.PHONY: build-kustomize-api
-build-kustomize-api: $(builtinplugins)
-	cd api; go build ./...
+.PHONY: test-unit-all
+test-unit-all: $(builtinplugins)
+	./hack/for-each-module.sh "make test"
 
-.PHONY: generate-kustomize-api
-generate-kustomize-api: $(MYGOBIN)/k8scopy
-	cd api; go generate ./...
+.PHONY: test-unit-non-plugin
+test-unit-non-plugin:
+	./hack/for-each-module.sh "make test" "./plugin/*" 15
+
+# This target is used by our Github Actions CI to run unit tests for all non-plugin modules in multiple GOOS environments.
+.PHONY: build-non-plugin-all
+build-non-plugin-all:
+	./hack/for-each-module.sh "make build" "./plugin/*" 15
 
 .PHONY: test-unit-kustomize-api
 test-unit-kustomize-api: build-kustomize-api
-	cd api; go test ./...  -ldflags "-X sigs.k8s.io/kustomize/api/provenance.version=v444.333.222"
-	cd api/krusty; OPENAPI_TEST=true go test -run TestCustomOpenAPIFieldFromComponentWithOverlays
+	cd api; $(MAKE) test
 
 .PHONY: test-unit-kustomize-plugins
 test-unit-kustomize-plugins:
 	./hack/testUnitKustomizePlugins.sh
 
 .PHONY: test-unit-kustomize-cli
 test-unit-kustomize-cli:
-	cd kustomize; go test ./...
+	cd kustomize; $(MAKE) test
 
 .PHONY: test-unit-kustomize-all
 test-unit-kustomize-all: \
 	test-unit-kustomize-api \
 	test-unit-kustomize-cli \
 	test-unit-kustomize-plugins
 
-test-unit-cmd-all:
+test-unit-kyaml-all:
 	./hack/kyaml-pre-commit.sh
```

Comment:
I think the idea of it was to cover what you might break when making changes to kyaml. CI isn't going to use it anymore, and it isn't actually true that you can't break other things. So I changed `verify-kustomize` to `verify-kustomize-repo` and used that to give us one big target we can run locally to be exhaustive. I dropped the kyaml and cmd/config test targets, since if that's what you want, you can just cd into those directories and run `make test` very easily. But the functions one still seemed useful, so I created it, but so that it picks up directories dynamically instead of based on a list now. LMK what you think.

Reply:
SGTM!